packages/components/src/components/Dropdown/Menu.tsx

```diff
@@ -0,0 +1,412 @@
+
+            :after {
+                position: absolute;
+                width: 100%;
+                top: -${spacingsPx.xs};
+                left: 0;
+                border-top: 1px solid ${theme.borderOnElevation1};
+                content: '';
+            }
+        `}
+`;
+
+interface AddonProps {
+    label: React.ReactNode;
+    icon: IconProps['icon'];
+    setToggled: (toggled: boolean) => void;
+    onClick?: () => void;
+}
+
+const Addon = ({
+    label,
+    icon,
+    isFocusable,
+    isKeyboardSelected,
+    onClick,
+    setToggled,
+}: AddonProps & { isFocusable?: boolean; isKeyboardSelected?: boolean }) => {
+    const theme = useTheme();
+
+    const addonRef = useRef<HTMLButtonElement>(null);
+
+    useEffect(() => {
+        if (isKeyboardSelected && addonRef.current) {
+            addonRef.current.focus();
+        }
+    }, [isKeyboardSelected]);
+
+    const handleAddonClick = () => {
+        if (onClick) {
+            onClick();
+            setToggled(false);
+        }
+    };
+
+    const onAddonKeyDown = (e: React.KeyboardEvent) => {
+        if (e.key === ' ' || e.key === 'Enter') {
+            e.preventDefault();
+            handleAddonClick();
+        }
+    };
+
+    return (
+        <AddonConteiner
+            tabIndex={0}
+            ref={addonRef}
+            onClick={handleAddonClick}
+            isFocusable={isFocusable}
+            onKeyDown={onAddonKeyDown}
+        >
+            <span>{label}</span>
+            <Icon icon={icon} size={spacings.sm} color={theme.iconPrimaryDefault} />
+        </AddonConteiner>
+    );
+};
+
+export interface DropdownMenuItemProps {
+    key: string;
+    label: React.ReactNode;
+    onClick?: () => any | Promise<any>;
+    shouldCloseOnClick?: boolean;
+    icon?: IconProps['icon'];
+    iconRight?: IconProps['icon'];
+    isDisabled?: boolean;
+    isHidden?: boolean;
+    separatorBefore?: boolean;
+    'data-test'?: string;
+}
+
+interface MenuItemComponentProps extends DropdownMenuItemProps {
+    isKeyboardSelected: boolean;
+    setToggled: (toggled: boolean) => void;
+}
+
+const MenuItem = ({
+    icon,
+    iconRight,
+    label,
+    isDisabled,
+    onClick,
+    shouldCloseOnClick = true,
+    setToggled,
+    isKeyboardSelected,
+    ...rest
+}: MenuItemComponentProps) => {
+    const itemRef = useRef<HTMLLIElement>(null);
+
+    useEffect(() => {
+        if (isKeyboardSelected && itemRef.current) {
+            itemRef.current.focus();
+        }
+    }, [isKeyboardSelected]);
+
+    const onMenuItemClick = (e: React.MouseEvent) => {
+        e.stopPropagation();
+
+        if (isDisabled) {
+            return;
+        }
+
+        onClick?.();
+        if (shouldCloseOnClick) {
+            setToggled(false);
+        }
+    };
+
+    const onMenuItemKeyDown = (e: React.KeyboardEvent) => {
+        if (e.key === ' ' || e.key === 'Enter') {
+            e.preventDefault();
+            onMenuItemClick(e as any);
```

Comment:
You can adjust the type on `onMenuItemClick` instead.

Reply:
0d684ed1d7